File: DSA/PriyankaSingh_590016827/day24question2.c
#include <stdio.h>
#include <stdlib.h>

// Node structure
struct Node 
{
    int data;
    struct Node* next;
};

// Create a new node
struct Node* createNode(int value)
{
    struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
    newNode->data = value;
    newNode->next = NULL;
    return newNode;
}

// Insert node at the end
struct Node* insertNode(struct Node* head, int value) 
{
    struct Node* newNode = createNode(value);
    if (head == NULL)
        return newNode;

    struct Node* temp = head;
    while (temp->next != NULL)
        temp = temp->next;

    temp->next = newNode;
    return head;
}

// Function to delete N nodes after skipping M nodes
void deleteNAfterM(struct Node* head, int M, int N) 
{
    struct Node* current = head;
    struct Node* temp;

    while (current) 
    {
        // Skip M nodes
        for (int i = 1; i < M && current != NULL; i++) 
        {
            current = current->next;
        }

        // If we've reached the end, break
        if (current == NULL)
            break;

        // Start deletion from next node
        temp = current->next;

        // Delete N nodes
        for (int i = 0; i < N && temp != NULL; i++)
        {
            struct Node* del = temp;
            temp = temp->next;
            free(del);
        }

        // Connect current to the node after the deleted ones
        current->next = temp;

        // Move current to next node for next iteration
        current = temp;
    }
}

// Function to print the linked list
void printList(struct Node* head) 
{
    struct Node* temp = head;
    while (temp != NULL) 
    {
        printf("%d -> ", temp->data);
        temp = temp->next;
    }
    printf("NULL\n");
}

// Main function
int main() 
{
    struct Node* head = NULL;
    int n, value, M, N;

    printf("Enter the number of nodes in the list: ");
    scanf("%d", &n);

    printf("Enter %d node values:\n", n);
    for (int i = 0; i < n; i++) 
    {
        scanf("%d", &value);
        head = insertNode(head, value);
    }

    printf("Enter M (nodes to skip): ");
    scanf("%d", &M);

    printf("Enter N (nodes to delete): ");
    scanf("%d", &N);

    printf("Original Linked List:\n");
    printList(head);

    deleteNAfterM(head, M, N);

    printf("Modified Linked List:\n");
    printList(head);

    return 0;
}

File: DSA/Nivedan_590014904/Day_24_Question_2.py
class Node:
    def __init__(self,data,next_node=None):
        self.data = data
        self.next = next_node
num = int(input("Enter the number of nodes to be put in the linked list : "))
head = None
current = None
for i in range(num):
    val = int(input(f"Enter the value of the {i+1} Node: "))
    new_node = Node(val)
    if head is None:
       head = new_node
       current = new_node
    else:
        current.next = new_node
        current = new_node
n  = int(input("Enter the number of nodes you want to delete(n): "))
m = int(input("Enter the number of nodes you want to skip(m): "))
p1 = head
p3 = head
if n+m < num:
    for i in range(m):
       p2 = p1
       p1=p1.next
    for i in range(n):
       p1 = p1.next
p2.next = p1
while p3 is not None:
    print(p3.data)
    p3 = p3.nextclass Node:
    def __init__(self,data,next_node=None):
        self.data = data
        self.next = next_node
num = int(input("Enter the number of nodes to be put in the linked list : "))
head = None
current = None
for i in range(num):
    val = int(input(f"Enter the value of the {i+1} Node: "))
    new_node = Node(val)
    if head is None:
       head = new_node
       current = new_node
    else:
        current.next = new_node
        current = new_node
n  = int(input("Enter the number of nodes you want to delete(n): "))
m = int(input("Enter the number of nodes you want to skip(m): "))
p1 = head
p3 = head
if n+m < num:
    for i in range(m):
       p2 = p1
       p1=p1.next
    for i in range(n):
       p1 = p1.next
p2.next = p1
while p3 is not None:
    print(p3.data)
    p3 = p3.next
